pyscriptjs/src/components/pyscript.ts

```diff
@@ -70,138 +70,52 @@ export function make_PyScript(interpreter: InterpreterClient, app: PyScriptApp)
-
-    // Keyboard Events
-    ['py-keydown', 'keydown'],
-    ['py-keypress', 'keypress'],
-    ['py-keyup', 'keyup'],
-
-    // Mouse Events
-    ['py-click', 'click'],
-    ['py-dblclick', 'dblclick'],
-    ['py-mousedown', 'mousedown'],
-    ['py-mousemove', 'mousemove'],
-    ['py-mouseout', 'mouseout'],
-    ['py-mouseover', 'mouseover'],
-    ['py-mouseup', 'mouseup'],
-    ['py-mousewheel', 'mousewheel'],
-    ['py-wheel', 'wheel'],
-
-    // Drag Events
-    ['py-drag', 'drag'],
-    ['py-dragend', 'dragend'],
-    ['py-dragenter', 'dragenter'],
-    ['py-dragleave', 'dragleave'],
-    ['py-dragover', 'dragover'],
-    ['py-dragstart', 'dragstart'],
-    ['py-drop', 'drop'],
-    ['py-scroll', 'scroll'],
-
-    // Clipboard Events
-    ['py-copy', 'copy'],
-    ['py-cut', 'cut'],
-    ['py-paste', 'paste'],
-
-    // Media Events
-    ['py-abort', 'abort'],
-    ['py-canplay', 'canplay'],
-    ['py-canplaythrough', 'canplaythrough'],
-    ['py-cuechange', 'cuechange'],
-    ['py-durationchange', 'durationchange'],
-    ['py-emptied', 'emptied'],
-    ['py-ended', 'ended'],
-    ['py-loadeddata', 'loadeddata'],
-    ['py-loadedmetadata', 'loadedmetadata'],
-    ['py-loadstart', 'loadstart'],
-    ['py-pause', 'pause'],
-    ['py-play', 'play'],
-    ['py-playing', 'playing'],
-    ['py-progress', 'progress'],
-    ['py-ratechange', 'ratechange'],
-    ['py-seeked', 'seeked'],
-    ['py-seeking', 'seeking'],
-    ['py-stalled', 'stalled'],
-    ['py-suspend', 'suspend'],
-    ['py-timeupdate', 'timeupdate'],
-    ['py-volumechange', 'volumechange'],
-    ['py-waiting', 'waiting'],
+// Differently from CSS selectors, XPath can crawl attributes by name and select
+// directly attribute nodes. This allows us to look for literally any `py-*` attribute.
+// TODO: could we just depend on basic-devtools module?
+// @see https://github.com/WebReflection/basic-devtools
+const $x = (path: string, root: Document | HTMLElement = document): (Node | Attr)[] => {
+    const expression = new XPathEvaluator().createExpression(path);
+    const xpath = expression.evaluate(root, XPathResult.ORDERED_NODE_SNAPSHOT_TYPE);
+    const result = [];
+    for (let i = 0, { snapshotLength } = xpath; i < snapshotLength; i++) {
+        result.push(xpath.snapshotItem(i));
+    }
+    return result;
+};
 
-    // Misc Events
-    ['py-toggle', 'toggle'],
-]);
+/** A weak relation between an element and current interpreter */
+const elementInterpreter: WeakMap<Element, InterpreterClient> = new WeakMap();
 
-/** Initialize all elements with py-* handlers attributes  */
+/** Initialize all elements with py-* handlers attributes */
 export function initHandlers(interpreter: InterpreterClient) {
     logger.debug('Initializing py-* event handlers...');
-    for (const pyAttribute of pyAttributeToEvent.keys()) {
-        createElementsWithEventListeners(interpreter, pyAttribute);
+    for (const { name, ownerElement: el } of $x('//@*[starts-with(name(), "py-")]') as Attr[]) {
```

Comment:
Do you think we should add a comment ? Using XPath is sometimes as mysterious as regex 😆

Reply:
Apologies I've read this only now ... so here the thing: we are using a copy and paste utility from the project linked right before such utility. The project readme explains why Xpath is cool and needed for basically this very same use case (except it checks for all `data-*` attributes) https://github.com/WebReflection/basic-devtools/#whats-the-deal-with-xpath

In there there is a link to XPath cheatsheet too ... I don't know how much value a comment would add in here because either it explains what's XPath and how it works or it points at some documentation like the README of the project does.

On that note, I suggest we use that module as dependency in case there's any other place we:
 * use `document.querySelector`
 * use `document.querySelectorAll`
 * use or need to use Xpath

That would simplify by far most of the code (at least it'll be easier on our eyes) and the module is 206 bytes so completely irrelevant in terms of code-size + as dependency is pretty much done, the *basic-devtools* is code covered 100% widely tested (we used that for years in ABP and other AdBlockers) and it's less to maintain or worry about in here.

Thoughts?

/cc @antocuni @JeffersGlass

Reply:
No worries and that makes sense, I never really grokked xpath so that's on me 😆 

Again, I haven't been in the loop on the current status of pyscript with all the work done since the Lisbon offsite, but I think we may be using `document.getElementById` a bunch of times, not sure about `querySelector` though